### PR TITLE
Re-add app engine deploy smoke tests

### PR DESCRIPTION
**Pull request by ipcrm**

### CODE REVIEW

File: lib/helpers.ts [hunk 1]
```diff
@@ -91,3 +93,60 @@ export const appEngineEphemeral = new Container({
...
+                        value: `https://${
+                            env === "production"
+                                ? configurationValue<string>("sdm.webapp.urls.prod")
+                                : configurationValue<string>("sdm.webapp.urls.staging")
+                        }`,
+                    },
+                ],
+                image: "cypress/included:6.1.0",
+                name: "cypress-included",
+                pushTest: and(repoSlugMatches(/^atomisthq\/web-ap-cljs$/), hasFile("cypress.json")),
```
Comment (ddgenome) [inline lib/helpers.ts:113]: Does the pushTest go here?

File: lib/helpers.ts [hunk 2]
```diff
@@ -91,3 +93,60 @@ export const appEngineEphemeral = new Container({
     .with(registration)
     .withProjectListener(appEngineVersioner)
     .withProjectListener(appEngineListener);
+
+export const [runSmokeTestStaging, runSmokeTestProduction] = ["staging", "production"].map(env =>
```
Comment (ddgenome) [inline lib/helpers.ts:97]: Aren't most (all?) of the other goals defined in the index.ts?

Comment (ipcrm) [reply lib/helpers.ts]: Moved to index.

File: lib/helpers.ts (hunk 1 above)
Comment (ipcrm) [reply lib/helpers.ts]: This is intentional.  I only want to have 1  `deployAppEngine` goal set and not have to have another one for when this file is present.  This way, if we have to shut it off again temporarily, we can just delete this file from the project.

File: index.ts [hunk 3]
```diff
@@ -484,6 +484,63 @@ export const configuration = configure(async sdm => {
...
+                            value: `https://${
+                                env === "production"
+                                    ? sdm.configuration.sdm.webapp.urls.prod
+                                    : sdm.configuration.sdm.webapp.urls.staging
+                            }`,
+                        },
+                    ],
+                    image: "cypress/included:6.1.0",
+                    name: "cypress-included",
+                    pushTest: and(repoSlugMatches(/^atomisthq\/web-ap-cljs$/), hasFile("cypress.json")),
```
Comment (ddgenome) [inline index.ts:503]: What I mean to ask is, does putting a push test here, inside the container structure, result in the provided push test being used when scheduling goals? I do not believe either the generic container or the k8s container interface has a pushTest property, so I do not think putting this here will have the effect you desire.

**Review by ddgenome — COMMENTED** [review]
I think one question remains.

File: index.ts (hunk 3 above)
Comment (ipcrm) [reply index.ts]: Ya, I don't know the answer to that.  Given that `Container` goals extend the `FulfillableGoalWithRegistrations` which ultimately uses `FulfillmentRegistration` I guess I just saw the type and wrongly assumed.  That said, I've removed it.  If we have a problem in the future we'll just have to remove the goal.

Comment (ddgenome) [reply index.ts]: I wasn't trying to say goals can't have push tests. Rather, I don't _think_ that is the correct place to put it.

**Review by ddgenome — APPROVED** [review]
LGTM

**Pull request merged**